homeassistant/components/sanix/sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Platform for Sensor integration."""
+from datetime import date, datetime
+
+from sanix.const import (
+    ATTR_API_BATTERY,
+    ATTR_API_DEVICE_NO,
+    ATTR_API_DISTANCE,
+    ATTR_API_FILL_PERC,
+    ATTR_API_SERVICE_DATE,
+    ATTR_API_SSID,
+    ATTR_API_TIME,
+)
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfLength
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import SanixCoordinator
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ATTR_API_BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_DISTANCE,
+        native_unit_of_measurement=UnitOfLength.CENTIMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_SERVICE_DATE,
+        device_class=SensorDeviceClass.DATE,
+        translation_key="service_date",
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_TIME,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        translation_key="time",
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_FILL_PERC,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key="fill_percentage",
+        icon="mdi:water-percent",
```

Comment:
Please use the new icon translations. (This uses the translation key to reference to icons, checkout integrations like `adguard` and others with an icons.json file on how that works)